### DISCUSSION TITLE
Vaillant protocol question

### DISCUSSION THREAD

**Discussion by rollercontainer** [john30/ebusd#658] [Q&A]
Hi,
I've got a Vaillant ecoTEC plus VC 206/5-5 and VRC470/4 mounted directly into the burner. I managed to get data via Ebus Adapter 1.6 atached to a WT32-ETH01 with Tasmota and SML decoder.
After dumping incoming data and comparing it to the protocol [vaillant_ebus_v0.6.0.pdf](https://ebus-wiki.org/lib/exe/fetch.php/ebus/vaillant_ebus_v0.6.0.pdf), I wasn't able to find data on the "right" places.

The "B5h 10h - Operational Data from Room Controller to Burner Control Unit" says that lead and service target temps are at byte 08 and 09, but in my data there is always 00 FF
`10 08 b5 10 09 00 00 00 ff ff ff 01 00 00 26 00 01 01 9a 00 `
Besides the B5h 10h 09h there is a second block with data length 03h:
`10 08 b5 10 03 05 ff 01 98 00 01 01 9a 00`
which isn't documented.

Does anyone have an idea how to decode my data?

**Comment by leonundjulie** [comment]
Good morning. I am looking to the feature "connecting WOLF Gastherme CGB-K 20" via ebus to my smarthome System. For several reasons, I am looking for a TASMOTA based solution. PLease be so kuind and provide some further information about your way to connect "Ebus Adapter 1.6 atached to a WT32-ETH01 with Tasmota and SML decoder."

**Comment by rollercontainer** [comment]
You have to compile the Tasmota esp32 firmware with Script support. There is no pre-compiled ethernet firmware with script support.
Connect the Ebus Adapter rx pin to a rx pin of the eth32. (yes, rx to rx...) In my case GPIO12. Then add your script. Wolf will have different IDs, so you have to find them yourself. You can dump all meter readings to the console (command Sensor53).

[https://tasmota.github.io/docs/Smart-Meter-Interface/#meter-definition](https://tasmota.github.io/docs/Smart-Meter-Interface/#meter-definition)


My Script:

```
>D
>B
=>sensor53 r
>M 3
+1,15,c,1,-50,Gas
+2,14,v,0,9600,Solar
+3,12,e,0,2400,Heizung
1,1-0:1.8.0*255(@100,Zählerstand,m³,total_in,2
3,xxxxB51009xxxxuu@2,Vorlauf soll,°C,lead target,1
3,xxxxB5110101xxxxxxuu@2,Vorlauf,°C,lead,1
3,xxxxB5110101xxxxxxxxuu@2,Rücklauf,°C,return,1
3,xxxxB5110101xxxxxxxxxxxxss@1,Aussen,°C,outside,0
3,xxxxB51009xxxxxxuu@2,Brauchwasser soll,°C,service_target,1
3,xxxxB5110101xxxxxxxxxxxxxxxxuu@2,Auslauf,°C,Auslauf,1
3,xxxxB5110101xxxxxxxxxxxxxxxxxxuu@2,Speicher,°C,speicher,1
3,xxxxB5110101xxxxxxxxxxxxxxxxxxxxuu@b0:1,Heizung,,heating,0
3,xxxxB5110101xxxxxxxxxxxxxxxxxxxxuu@b1:1,Brauchwasser,,service_water,0 
#
```

**Reply by leonundjulie** [reply]
Thanks for your answer. I compiled already a TASMOTA package including script support (was necessary due to the parallel task "Read out electricity meter". By means - on a global point of view - if I do it like you:
- take an eBus adapter 1.6 (or newer),
- connect the EBus adapter to my ESP8266
- TASMOTA is flashed on the ESP8266.
- the "TASMOTA bin" is including script support

... then I only have to lay out the script according to WOLF ;-)

Is the following short version a working thesis: eBus adapter + ESP8266 + TASMOTA on WOLF will work?

Then the TASMOTA flashed ESP8266 would send everything via MQT into the ioBroker and I already have what I want to achieve (all heating data in my Smarthome platform).

**Comment by rollercontainer** [comment]
you have to compile SML support too: https://tasmota.github.io/docs/Smart-Meter-Interface/
In my case, I am only able to catch the broadcasts from the ebus. If I add the optocoppler for the writing/TX path my ebus tilts and the display goes down. That way I am not able to ask for specific data or set variables...

All in all I would not suggest following that path.

Another way for Vaillant boilers is to buy a opentherm board (<100€ in the Netherlands) and talk via a simple pcb with a Arduino or ESP. 
https://myboiler.com/opentherm/vaillant-opentherm/
https://tasmota.github.io/docs/OpenTherm/
Maybe Wolf can talk Opentherm too? Opentherm is not as closed as ebus because it is, you guessed it, open... :-)
